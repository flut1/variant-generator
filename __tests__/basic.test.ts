import createVariants from '../src/index';

describe('createVariants()', () => {
  test('handles simple 2x2 input', () => {
    const result = createVariants(function*() {
      return {
        bool: yield [false, true],
        num: yield [0, 1],
      };
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
  test('handles simple 3x2 input', () => {
    const result = createVariants(function*() {
      return [yield [0, 1], yield [2, 3], yield [4, 5]];
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
  test('works with a generator without yields', () => {
    const result = createVariants(function*() {
      return [1, 2, 3];
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
  test('creates different variations for deep-equal inputs', () => {
    const result = createVariants(function*() {
      return [yield [0, 1, 2], yield [0, 1, 2]];
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
  test(`doesn't duplicate variations for strictly equal inputs`, () => {
    const input = [0, 1, 2];
    const result = createVariants(function*() {
      return [yield input, yield input];
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
});
