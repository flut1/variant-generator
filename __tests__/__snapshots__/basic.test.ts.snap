// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createVariants() creates different variations for deep-equal inputs 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    0,
    1,
  ],
  Array [
    0,
    2,
  ],
  Array [
    1,
    0,
  ],
  Array [
    1,
    1,
  ],
  Array [
    1,
    2,
  ],
  Array [
    2,
    0,
  ],
  Array [
    2,
    1,
  ],
  Array [
    2,
    2,
  ],
]
`;

exports[`createVariants() doesn't duplicate variations for strictly equal inputs 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    1,
    1,
  ],
  Array [
    2,
    2,
  ],
]
`;

exports[`createVariants() handles simple 2x2 input 1`] = `
Array [
  Object {
    "bool": false,
    "num": 0,
  },
  Object {
    "bool": false,
    "num": 1,
  },
  Object {
    "bool": true,
    "num": 0,
  },
  Object {
    "bool": true,
    "num": 1,
  },
]
`;

exports[`createVariants() handles simple 3x2 input 1`] = `
Array [
  Array [
    0,
    2,
    4,
  ],
  Array [
    0,
    2,
    5,
  ],
  Array [
    0,
    3,
    4,
  ],
  Array [
    0,
    3,
    5,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
    5,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
    5,
  ],
]
`;

exports[`createVariants() works with a generator without yields 1`] = `
Array [
  Array [
    1,
    2,
    3,
  ],
]
`;
